### PR TITLE
Replace knox with aws-sdk

### PR DESCRIPTION
**Pull request by brymut**
Replace the no longer maintained knox library with aws-sdk. Actively maintained by AWS over at https://github.com/aws/aws-sdk-js

### CODE REVIEW

**Comment by brymut** [comment]
CC :  opencollective/opencollective#1880

**Comment by brymut** [comment]
Seems to be working fine against some manual local tests & the repo's unit tests I ran against a personal AWS bucket

File: package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "apollo-server-express": "2.5.0",
     "argparse": "1.0.10",
     "async": "2.6.1",
+    "aws-sdk": "^2.460.0",
```
Comment (znarf) [inline package.json]: We use fixed dependencies in this repository.

File: server/lib/awsS3.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import config from 'config';
+
+// Create S3 service object & set credentials and region
+let s3;
+if (config.aws.s3.key) {
+  s3 = new S3({
+    accessKeyId: config.aws.s3.key,
+    secretAccessKey: config.aws.s3.secret,
+    apiVersion: '2006-03-01',
+    region: 'us-west-1',
```
Comment (znarf) [inline server/lib/awsS3.js]: Let's make it nice and have `apiVersion` and `region` configurable.

Add an environment variable here:
- https://github.com/opencollective/opencollective-api/blob/master/config/custom-environment-variables.json

And current values as default here:
- https://github.com/opencollective/opencollective-api/blob/master/config/default.json

**Comment by znarf** [comment]
Looks good to me. Just have small feedback. We should be able to merge tomorrow.

File: package.json (hunk 1 above)
Comment (brymut) [reply package.json]: ahh, completely slipped my mind, will commit the correction. Would it be worth putting this as a lint rule or something in the future? More than happy to create an issue and look into it when I can.

File: server/lib/awsS3.js (hunk 2 above)
Comment (brymut) [reply server/lib/awsS3.js]: good point, I'll get on this

File: package.json (hunk 1 above)
Comment (znarf) [reply package.json]: I don't know something simple that would do the trick. Not sure it's worth it if it's complicated.

Proposals welcome!

Comment (brymut) [reply package.json]: Alright, I'll have a look around soon and will create a new issue & let you know if I find something worthwhile.

**Comment by brymut** [comment]
Hi @znarf , made the small changes. Should be good now.

File: config/custom-environment-variables.json [hunk 3]
```diff
@@ -52,7 +52,9 @@
     "s3": {
       "key": "AWS_KEY",
       "secret": "AWS_SECRET",
-      "bucket": "AWS_S3_BUCKET"
+      "bucket": "AWS_S3_BUCKET",
+      "region": "AWS_REGION",
```
Comment (znarf) [inline config/custom-environment-variables.json]: `AWS_S3_REGION`

File: config/custom-environment-variables.json [hunk 4]
```diff
@@ -52,7 +52,9 @@
...
+      "apiVersion": "AWS_API_VERSION"
```
Comment (znarf) [inline config/custom-environment-variables.json]: `AWS_S3_API_VERSION`

Comment (brymut) [reply config/custom-environment-variables.json]: damn, yep, missed that

**Comment by brymut** [comment]
Hi again, fixed the naming mistakes. More than happy to squash these commits if that's alright with you?

**Comment by brymut** [comment]
Hi @znarf, submitted an expense for this. https://opencollective.com/engineering/expenses/8176

**Pull request merged**